src/sentry/integrations/vsts/webhooks.py

```diff
@@ -44,9 +46,19 @@ def check_webhook_secret(self, request, integration):
 
     def handle_updated_workitem(self, data, integration):
         external_issue_key = data['resource']['workItemId']
-        assigned_to = data['resource']['fields'].get('System.AssignedTo')
-        status_change = data['resource']['fields'].get('System.State')
         project = data['resourceContainers']['project']['id']
+        try:
+            assigned_to = data['resource']['fields'].get('System.AssignedTo')
+            status_change = data['resource']['fields'].get('System.State')
+        except KeyError:
+            logger.info(
+                'vsts.updated-workitem-fields-not-passed',
+                extra={
+                    'payload': data,
```

Comment:
after merging, can you keep an eye on this and see if there are certain fields we can pull out so we don't have to log the whole payload?

i realize that since this is the webhook behaving unexpectedly you probably don't have a good idea now, but it'd be nice to not have to log the whole blob forever -- stuff like account, etc would be good things to log

Reply:
That's really good insight! Definitely.